### ISSUE TITLE
Windows Service Fails Sporadically To Update Slack Status 

### ISSUE THREAD

**Issue by fanboyfanboy** [tharakadesilva/SlackUserPresenceManager#17]
I've noticed that sometimes when logging into my machine after it had been locked, the Windows service will fail to update my Status for >=1 team(s). Unsure of the root cause of this - will need to investigate further.

info: 
Windows 7
Slack Desktop App - latest build
SlackUserPresenceManager - v1.1

**Comment by tharakadesilva** [comment]
Actually this happens to me when I am not connected to the internet when I log into the computer. The first one will take some time and timeout. That timeout period will allow the internet to be reconnected and for the second request to go through. We should look into adding a retry mechanism to this.

**Comment by fanboyfanboy** [comment]
agreed a retry mechanism would be solid. Also though I should note this has also happened to me in the middle of the work day when I wouldn't think I lost internet connection
